### PR TITLE
fix build errors

### PR DESCRIPTION
**Pull request by yajiefeng**

### CODE REVIEW

File: beat/redisbeat_test.go [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-	cfg "github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestDefaultConfig(t *testing.T) {
 	// modify cfgfile's default location
-	os.Args = []string{"../redisbeat.yaml"}
-	cfg.ChangeDefaultCfgfileFlag("redisbeat")
+	//os.Args = []string{"../redisbeat.yaml"}
```
Comment (chrsblck) [inline beat/redisbeat_test.go]: These two commented out lines can be removed, right?

File: beat/redisbeat.go [hunk 2]
```diff
@@ -156,27 +161,28 @@ func (rb *Redisbeat) Config(b *beat.Beat) error {
...
+	logp.Debug("redisbeat", "Client statistics %t", rb.clientsStats)
+	logp.Debug("redisbeat", "Memory statistics %t", rb.memoryStats)
+	logp.Debug("redisbeat", "Persistence statistics %t", rb.persistenceStats)
+	logp.Debug("redisbeat", "Stats statistics %t", rb.statsStats)
+	logp.Debug("redisbeat", "Replication statistics %t", rb.replicationStats)
+	logp.Debug("redisbeat", "Cpu statistics %t", rb.cpuStats)
+	logp.Debug("redisbeat", "Command statistics %t", rb.commandStats)
+	logp.Debug("redisbeat", "Cluster statistics %t", rb.clusterStats)
+	logp.Debug("redisbeat", "Keyspace statistics %t", rb.keyspaceStats)
+	logp.Debug("redisbeat", "RequiredPass %v", rb.pass)
```
Comment (chrsblck) [inline beat/redisbeat.go]: I'd rather not echo back the password, even though it's stored in a config file somewhere.

File: beat/redisbeat_test.go [hunk 3]
```diff
@@ -40,11 +44,17 @@ func TestDefaultConfig(t *testing.T) {
 
 func TestModifiedConfig(t *testing.T) {
 	// modify cfgfile's default location
-	os.Args = []string{"../tests/redisbeat.yml"}
-	cfg.ChangeDefaultCfgfileFlag("redisbeat")
+	//os.Args = []string{"../tests/redisbeat.yml"}
```
Comment (chrsblck) [inline beat/redisbeat_test.go]: Same for these two lines...

File: beat/redisbeat.go [hunk 4]
```diff
@@ -41,13 +40,19 @@ type Redisbeat struct {
...
+
+	return rb, err
 }
 
-func (rb *Redisbeat) Config(b *beat.Beat) error {
+func (rb *Redisbeat) config(b *beat.Beat, cfg *common.Config) error {
 
-	err := cfgfile.Read(&rb.RbConfig, "")
+	err := cfg.Unpack(&rb.RbConfig)
+	//err := cfgfile.Read(&rb.RbConfig, "")
```
Comment (chrsblck) [inline beat/redisbeat.go]: This should be removed if it's not needed.

File: redisbeat.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@ input:
   #period: 10
 
   # Host of redis server, default is localhost
-  #host: "localhost"
+  host: "localhost"
```
Comment (chrsblck) [inline redisbeat.yml]: Why are we uncommenting these? Do the defaults in code (`redisbeat.config()`) not work anymore?

**Comment by chrsblck** [comment]
@jieniu Thanks for this Pull Request!

File: redisbeat.yml (hunk 5 above)
Comment (yajiefeng) [reply redisbeat.yml]: yes, it didn't work with the latest beatlib

Comment (chrsblck) [reply redisbeat.yml]: Do you happen to know why the following doesn't work now? 

```
if rb.RbConfig.Input.Host != nil {
	rb.host = *rb.RbConfig.Input.Host
} else {
	rb.host = DEFAULT_HOST
}
```

File: beat/redisbeat.go (hunk 2 above)
Comment (yajiefeng) [reply beat/redisbeat.go]: i just do some test, cause old code don't read the right passwd from configure file, i'll remove it later

File: beat/redisbeat.go (hunk 4 above)
Comment (yajiefeng) [reply beat/redisbeat.go]: yes, i'll remove it

File: beat/redisbeat_test.go (hunk 1 above)
Comment (yajiefeng) [reply beat/redisbeat_test.go]: already modified

File: redisbeat.yml (hunk 5 above)
Comment (yajiefeng) [reply redisbeat.yml]: it's ok, you can check it out again.

**Comment by chrsblck** [comment]
@jieniu Looks good. Can you squash all the commits into one before I merge it.

**Comment by yajiefeng** [comment]
@chrsblck, i'll try it later.

**Pull request closed without merging**